client/jetpack-connect/index.js

```diff
@@ -28,16 +32,18 @@ export default function() {
 	);
 
 	if ( config.isEnabled( 'jetpack/connect/remote-install' ) ) {
+		// This route is subject to a page redirect
+		// See `componentWillMount()` in client/jetpack-connect/remote-credentials.js
 		page(
-			'/jetpack/connect/install',
+			`/jetpack/connect/install/:locale${ lang }?`,
```

Comment:
Would it be possible to extract appending `:locale${ lang}` out? There are so many in this file so it would be great if we could think of a way to honor the DRY principle. 

Since we've been introducing locale-aware paths more and more, maybe a utility function like `createLocalizedPath()` in `i18n-util` which appends the locale path parameter format string properly?

Reply:
> Since we've been introducing locale-aware paths more and more, maybe a utility function like createLocalizedPath() in i18n-util which appends the locale path parameter format string properly?

It's a fine idea, but I think this is out of scope for what this PR wants to achieve. 

Particularly because some route controllers are expecting  the name `lang` and others `locale` for this fragment. It'd be great to unify this, you're right, but my opinion is that it's better to create a new PR so we can focus on making sure the abstraction works independently of any changes introduced here.

Reply:
Forgot to mention, I'll add it to the list of TODOs thought 😉

Reply:
Would it make sense to use the new `localizeUrl`? #27857

Reply:
We don't need to localize the route per se, but add a list of possible values for the `:locale` route fragment, which is basically all of our valid language slugs. :)

So, the value of `lang` in this literal `/:locale${ lang }?` is `(${ getLanguageSlugs().join( '|' ) })?`, which is `(ar|af|en|de...(and the rest))?`